### DISCUSSION TITLE
External URL with custom script

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [tauri-apps/tauri#4822] [Q&A] [answered]
I'm sure this is probably explained somewhere, but I was not able to find it.

The goal is to have Tauri opening an external website while loading a local javascript file on it. (Like a '[preload](https://www.electronjs.org/docs/latest/tutorial/tutorial-preload)' in Electron)

Thanks everyone in advance.

**Comment by FabianLars** [comment, marked as answer]
That's only possible if you create the window in rust and via this method: <https://docs.rs/tauri/latest/tauri/window/struct.WindowBuilder.html#method.initialization_script>

**Reply by MrAmericanMike** [reply]
Thanks a lot got it working.
Only thing I noticed is that for some reason the script triggers 2 or 3 times, depending on what website is open. I guess I will need to implement something to make sure it triggers only once.
Other than that issue, looks like it will do the trick.

**Reply by mindplay-dk** [reply]
is there a feature exposing this via configuration now, some years later? I don't have any working knowledge of Rust. 😌

**Reply by FabianLars** [reply]
no

**Reply by FabianLars** [reply]
a config would be the same as a rust api on a security basis imo though so please feel free to open a feature request issue (just don't expect a js api).
